agent/proxycfg/naming.go

```diff
@@ -13,6 +13,13 @@ type UpstreamID struct {
 	structs.EnterpriseMeta
 }
 
+func (u UpstreamID) Matches(other UpstreamID) bool {
```

Comment:
If you normalize the entmeta, you should be able to do just `==` on the `UpstreamID` instead of using a whole other method. If you needed this it probably means something else is subtly wrong, since UpstreamID has to be used as a map key in various places which requires it being normalized early.

Reply:
Ah nope didn't need that. Removed.